### DISCUSSION TITLE
[5.x] Account and entry actions UX

### DISCUSSION THREAD

**Discussion by andrewhawkes** [craftcms/cms#14913] [Ideas] [answered]
This is just an observation, and I am sharing my opinion on the UX of account actions in Craft CMS 5.
I'm not saying this is wrong, but it feels like a downgrade UX-wise to me.

I needed to send an activation email, and it took me quite a long time to find these actions:

<img width="494" alt="image" src="https://github.com/craftcms/commerce/assets/4975742/3bc0c61c-1c3c-4451-87dd-1a19966a5e0e">

These are very common actions, and they shouldn't be hidden behind a nondescript menu icon that could easily be confused for an ellipsis on the breadcrumb and missed.

I feel like there should be a clear menu button in the sidebar or other prominent area with a labelled button.

The same is true for deleting entries:

<img width="487" alt="image" src="https://github.com/craftcms/commerce/assets/4975742/73934649-ada3-4e1e-bd68-dd0ac6c1b962">

I would expect the delete action to be near the save action.

**Comment by MattWilcox** [comment]
Agreed - It's not a good UI, for multiple reasons.

1. It's location is abnormal. People are not looking there to find controls for things you can do on the page. Every other control is elsewhere. Tangentially - this is also why we have clients that have no idea they're editing a draft and haven't saved and that's why the live site isn't showing their edits. Had to video conference that very issue yesterday. Breadrumbs should not tag-team UI duties.
2. It's an a11y issue becuase it is essentially un-labeled. Icons alone are never good UI - that's a cognitive accessibility issue. That's why things should have text labels *and* icons. Icons on their own should only be used when an audience are sure to be familiar with the specific icons (think road signs... you have to take a test to be allowed to drive a car, and part of the test is "what's this icon mean"? Loads of people fail that, even when they've studied the book with the icons. Icons on their own are *bad UI* for a general audience.)
3. It's hiding important actions, and there just isn't any need to hide those actions.

Honestly feels like design decisions "to fit it into the breadcrumb", but that itself was where it went wrong IMO.

**Comment by andrewhawkes** [comment]
Just for reference, in Craft CMS 4 the delete action was paired with the save actions
<img width="268" alt="image" src="https://github.com/craftcms/cms/assets/4975742/3a2f8b07-c21f-4091-8383-7f84500f5802">

User actions were behind a much more obvious settings cog icon button:
<img width="373" alt="image" src="https://github.com/craftcms/cms/assets/4975742/6d6cb62a-d313-4a05-a60e-148fe4f9a3f8">

**Comment by vardumper** [comment]
The first spot I was looking to activate a user was on the index page of users (selecting one or many gave me dropdown with grayed out options suspend, etc.)
The second place I looked, was the left sidebar menu of a user profile (permissions, and more).

Let me add, that I would also have expected a Form to reset a Users' Password in a users' profile as well.

**Comment by thupsi** [comment]
I agree with all of the above. That breadcrumb menu icon is very easy to miss or mistake for a non-functional ellipsis, due to it's placement and styling.

**Comment by splendidrob** [comment]
Thanks so much for this! Took me ages to find that menu! One other small issue I noticed which feels like a downgrade from craft 4 is removing assets/entries from and asset or entry field. Its now a two click process to open the menu and click remove where as it used to just be a single click cross...this is pretty annoying when you have 30 assets or so in a gallery and need to remove them all...maybe there is a better way!?
<img width="339" alt="screenshot_1464" src="https://github.com/craftcms/cms/assets/25520763/8f90f5a3-04b2-46ce-abcc-6b70c6b68a72">

**Reply by brandonkelly** [reply]
Create a new issue for this: #15078

**Comment by andrewhawkes** [comment]
I've just had a client training session over a video call, and they found this really complicated.
We've had to write up documentation on registering a user and sending an activation email.

**Comment by mmikkel** [comment]
IMO, the UX when creating new users from the CP feels pretty broken on Craft 5 and the fact that those activation actions are a bit hidden away in that breadcrumb disclosure menu is only part of the problem.

I think the biggest issue is that the profile info and permissions have been moved into dedicated edit pages (along with addresses). Since changes in these edit pages isn't persisted between pageloads, the new user has to be saved (or "created") after entering info in each one. 

Compared to the tabbed interface in previous versions of Craft, this implementation makes the previously straight-forward _enter name and email, assign permissions, send activation email_ workflow a lot more complicated.

Crucially, if the author isn't a power user and clicks the primary Create/Save button instead of Cmd+S (or the "Create and continue editing" action), _they will be redirected back to the user index after each of those saves_, and have to track down the new user in the index to continue "creating" it. At this point, it's not unlikely that the state of the new user is somewhat unclear to the author, and it's also not guaranteed that the user is even included in the specific index source the author is redirected back to. This can be the experience even more confusing.

Finally, after saving the user 2 or 3 times, the author has to remember to send an activation email, and where that action is found – which I agree is probably not in the most intuitive spot 🙂

**I do think the new UI works well enough for editing existing users**, but hopefully the flow for creating new users specifically, can be improved.

Adding auto-save capabilities to the user edit pages, as well as making the activation actions more easily discoverable – or, have them available _in some form_ right inside the editing view when creating new users, similarly to the "Send an activation email now" checkbox in Craft 4 – would probably help a lot.

**Reply by brandonkelly** [reply]
Created a new issue for this: #15079

**Comment by brandonkelly** [comment, marked as answer]
We’ve moved the action menu into the page toolbar for Craft 5.2, where it should be a bit more discoverable. (#15070)

**Reply by andrewhawkes** [reply]
Thank you - this should make it a lot easier!

**Reply by brandonkelly** [reply]
Craft 5.10.0-beta.1 is out with that change.
